cep-sigstore-predicate.md

```diff
@@ -12,61 +12,182 @@
+
+However, this metadata is not currently **cryptographically verifiable**:
+the consuming party must either trust it as presented, or verify it manually
+against independent sources of truth (such as a project's release history).
+
+Attestations that present this metadata in a cryptographically
+verifiable manner are desirable for a number of reasons:
+
+* Package maintainers wish to demonstrate the integrity and authenticity
+  of their package uploads;
+* Individual downstream users wish to verify the integrity and authenticity of
+  packages they consume, without placing additional trust in the
+  channel or channel's hosting server;
+* Attestations change the sophistication and risk profile for attackers in
+  defenders' favor: the attacker must be sufficiently sophisticated
+  to access private key material, *and* have a risk tolerance profile that
+  accepts exposure via auditable transparency logs.
+
+More broadly, attestation schemes like the one proposed in this CEP have
+seen adoption in similar and related ecosystems:
+
+* Python (PyPA/PyPI): [PEP 740] and [PyPI - Attestations]
+* NodeJS (npm): [npm - Generating provenance statements]
+* Ruby (RubyGems): [rubygems/release-gem]
 
-### Sigstore Attestations
-
-Sigstore attestations are cryptographic statements about software artifacts that provide:
-
-- Authenticity: Proof of who created/signed the artifact
-- Integrity: Verification that the artifact hasn't been tampered with
-- Transparency: Public record of signatures in a tamper-evident log
-
-### Key Components
+## Specification
 
-- Predicates: JSON documents containing metadata about the signing event, using the `in-toto` format
-- Signatures: Cryptographic proofs made using ephemeral keys
-- Rekor: A tamper-evident log that stores attestations
-- Fulcio: A certificate authority that issues short-lived certificates
+### Attestation format
 
-In this document, we want to standardize the sigstore predicate for conda packages. The bundle format to be used for sigstore attestations is the `v0.3` bundle format.
+This CEP proposes the following attestation statement layout, using the
+[in-toto Statement schema]:
 
-## Specification
+- `predicateType` **MUST** be `https://schemas.conda.org/publish-v1.json`
+- `subject` **MUST** be a single [`ResourceDescriptor`], with the following
+  constraints:
+    - `subject[0].name` **MUST** be the full filename of the conda package
+      that will be part of the `repodata.json` and under which it will appear on
+      the server.
+    - `subject[0].digest` **MUST** be a [`DigestSet`], and it **MUST** contain
+      a single `sha256` entry with the SHA256 hash of the conda package.
+- `predicate` **MAY** be present. If present and not `null`, it **MUST** be a
+  JSON object with the following fields:
+    - `targetChannel` **MUST** be a string, indicating where the package
+      is being uploaded to. This field **MUST** be a valid URL with no
+      trailing slashes.
```

Comment:
Flagging: the original language said "canonical URL" for the `targetChannel`, but I couldn't find a clear resource on what canonicalization of channel URLs looks like in the conda ecosystem. Does conda have a definition for a "canonical" URL, or by "canonical" did you mean something like "official"?